app/src/main/java/com/google/samples/apps/sunflower/PlantListFragment.kt

```diff
@@ -70,7 +71,11 @@ class PlantListFragment : Fragment() {
 
     private fun subscribeUi(adapter: PlantAdapter) {
         viewModel.getPlants().observe(viewLifecycleOwner, Observer { plants ->
-            if (plants != null) adapter.submitList(plants)
+            if (plants != null) {
+                adapter.submitList(plants.map {
```

Comment:
The item view-model isnot created by `ViewModelProviders` , I do 
 prefe to `new` directly. 

Almost same case `PlantAndGardenPlantingsViewModel`. If you really want to let each item lifecycle sensitively, you can let all items `LifecycleObserver`. I use those pattern in MVVM-driven project.

Reply:
Yes, it seems that having a factory for the view model of the item of a list is a bit useless. I have updated the code for removing the factoring and instantiating the view model directly.